scenarios/pravega-flink-autoscaling/README.md

```diff
@@ -0,0 +1,183 @@
+
+
+### 4. Run Data Generator
+
+Next, we run the `Data Generator` application. Recall to update the
+`dataGenerator.yaml` file for pointing to the Docker image you have
+built in the first step. Also, you can parameterize some aspects of the
+application, such as the stream and scope names, the `Pravega` controller
+endpoint, as well as the stream scaling policy rate in events.
+
+- The `Data Generator` application can be deployed as follows:
+```
+cd data-generator
+kubectl apply -f dataGenerator.yaml
+```
+
+### 5. Deploy Flink Job
+It is time now to deploy the `Flink Application` processing stream
+data. Similar to the previous step, recall to update the Docker
+image in both `jobmanager-application.yaml` and `taskmanager-job-deployment.yaml`
+with the images you have built. You can also parameterize the `Flink Application`
+with the `Pravega` endpoint, as well as the computation load you want to
+simulate:
+```
+cd flink-application
+kubectl apply -f flink-configuration-configmap.yaml
+kubectl apply -f jobmanager-application.yaml
+kubectl apply -f jobmanager-service.yaml
+kubectl apply -f taskmanager-job-deployment.yaml
+```
+We provide a couple of scripts for starting up and tiering down `Flink` with
+a single command.
+
+### 6. Deploy Kubernetes Pod Auto-scaler
+
+- Finally, we need to integrate the Kubernetes `Horizontal Pod Auto-scaler`
+in the deployment. To do so, we first need to install `Prometheus Adapter`:
+```
+helm repo add prometheus-community https://prometheus-community.github.io/helm-charts
+helm repo update
+helm install prom-adap prometheus-community/prometheus-adapter -f ./values-prometheus-adapter.yaml
+```
+Note that in the `values-prometheus-adapter.yaml` we are instructing the `Prometheus
+Adapter` to exposing the "segment count" metric for a particular stream: 
+`'pravega_controller_segments_count{scope="test",stream="stream",}'`. If you
+want to use a different stream or a different metric for triggering
+auto-scaling actions, you will need to modify this file.
+
+- Finally, we deploy the Kubernetes `Horizontal Pod Auto-scaler` to auto-scale
+the number of `Flink Task Managers` based on the number of stream segments for this
+demo:
+```
+cd hpa-demo
+kubectl apply -f hpa-demo.yaml
+```
+
+As a result, you should be able to create some dashboards in `Grafana`
+to inspect both `Pravega` and `Flink` metrics. At the same time,
+you should be able to see a sinusoidal workload being generated by
+the `Data Generator` and the number of stream segments changing 
+as a result of the change in load and the stream scaling
+policy configured. Finally, this should lead to the Kubernetes
+`Horizontal Pod Auto-scaler` changing the number of `Flink Task Managers`
+according to the number of parallel segments in the stream.
+In the following, we provide some `Grafana` dashboard screenshots
+demonstrating this:
+
+![image](img/sinusoidal-workload.png)
```

Comment:
Is it good to show this image earlier in the `Data Generator` section to show the pattern of the load?

Reply:
I think that is better to keep the results for the last part, as we focus first on having the demo up and runnig.